File: My-Blog-Go/models/resource.go
package models

import (
	"time"
)

// Resource 资源信息表
type Resource struct {
	ID         uint      `gorm:"primaryKey;column:id" json:"id"`
	UserID     uint      `gorm:"column:user_id_id" json:"userId"`
	Type       string    `gorm:"column:type;size:32" json:"type"`
	Path       string    `gorm:"column:path;size:256" json:"path"`
	Size       *int      `gorm:"column:size" json:"size,omitempty"`
	MimeType   *string   `gorm:"column:mime_type;size:256" json:"mimeType,omitempty"`
	Status     bool      `gorm:"column:status;default:true" json:"status"`
	CreateTime time.Time `gorm:"column:create_time;autoCreateTime" json:"createTime"`
}

func (Resource) TableName() string {
	return "resource"
}

// ResourcePath 资源路径表
type ResourcePath struct {
	ID           uint      `gorm:"primaryKey;column:id" json:"id"`
	Title        string    `gorm:"column:title;size:64" json:"title"`
	Classify     *string   `gorm:"column:classify;size:32" json:"classify,omitempty"`
	Cover        *string   `gorm:"column:cover;size:256" json:"cover,omitempty"`
	URL          *string   `gorm:"column:url;size:256" json:"url,omitempty"`
	Introduction *string   `gorm:"column:introduction;size:1024" json:"introduction,omitempty"`
	Type         string    `gorm:"column:type;size:32" json:"type"`
	Status       bool      `gorm:"column:status;default:true" json:"status"`
	Remark       *string   `gorm:"column:remark;type:text" json:"remark,omitempty"`
	CreateTime   time.Time `gorm:"column:create_time;autoCreateTime" json:"createTime"`
}

func (ResourcePath) TableName() string {
	return "resource_path"
}

File: My-Blog-Vue3/src/types/resource.ts
/**
 * 资源相关类型定义
 */

/** 资源信息 */
export interface Resource {
  id: number;
  userId: number;
  /** 上传者用户名 */
  username: string;
  /** 资源类型：userAvatar, articleCover, articlePicture 等 */
  type: string;
  path: string;
  /** 文件大小 */
  size: number | null;
  /** MIME 类型 */
  mimeType: string | null;
  status: boolean;
  createTime: string;
}

/** 获取资源列表参数 */
export interface GetResourceListParams {
  current?: number;
  size?: number;
  /** 资源类型筛选（按 mime_type） */
  resourceType?: string;
  isAdmin?: boolean;
  noPagination?: boolean;
}

/** 修改资源状态参数 */
export interface ChangeResourceStatusParams {
  id: number;
  flag: boolean;
}

/** 保存资源参数 */
export interface SaveResourceParams {
  type: string;
  path: string;
  size: number;
  mimeType: string;
}

/** 资源路径信息 */
export interface ResourcePath {
  id: number;
  title: string;
  /** 分类 */
  classify: string | null;
  /** 封面 */
  cover: string | null;
  /** 链接 */
  url: string | null;
  /** 简介 */
  introduction: string | null;
  /** 资源类型：friendUrl, movie, music, wallpaper, favorites 等 */
  type: string;
  /** 友链头像 */
  friendAvatar: string | null;
  status: boolean;
  createTime: string;
}

/** 获取资源路径列表参数 */
export interface GetResourcePathListParams {
  current?: number;
  size?: number;
  /** 资源类型筛选 */
  resourceType?: string;
  /** 分类筛选 */
  classify?: string;
  isAdmin?: boolean;
  noPagination?: boolean;
}

/** 保存资源路径参数 */
export interface SaveResourcePathParams {
  title: string;
  classify?: string;
  cover?: string;
  url?: string;
  introduction?: string;
  type: string;
  friendAvatar?: string;
}

/** 更新资源路径参数（管理员） */
export interface UpdateResourcePathParams {
  id: number;
  title?: string;
  classify?: string | null;
  cover?: string | null;
  url?: string | null;
  introduction?: string | null;
  type?: string;
  remark?: string | null;
  friendAvatar?: string | null;
  status?: boolean;
}
